tests/meta/transfer/test_external_data.py

```diff
@@ -3,266 +3,150 @@
-    analysis_store_trio,
-    customer_id: str,
-    external_data_api: ExternalDataAPI,
-    sample_obj: models.Sample,
-    ticket: str,
-    tmpdir_factory,
-):
-    # GIVEN one such sample exists
-    tmp_dir_path: Path = Path(tmpdir_factory.mktemp(sample_obj.internal_id, numbered=False)).parent
-    available_samples = external_data_api.get_available_samples(folder=tmp_dir_path, ticket=ticket)
-    # THEN the function should return a list containing the sample object
-    assert available_samples == [sample_obj]
-
-
-def test_curate_sample_folder(
-    case_id, customer_id, dna_case, external_data_api: ExternalDataAPI, tmpdir_factory
+def test_start_associated_case(
+    external_data_handler: ExternalDataHandler, sample_obj: models.Sample
 ):
-    cases = external_data_api.status_db.query(models.Family).filter(
-        models.Family.internal_id == case_id
-    )
-    sample: models.Sample = cases.first().links[0].sample
-    tmp_folder = Path(tmpdir_factory.mktemp(sample.name, numbered=False))
-    external_data_api.curate_sample_folder(
-        cust_name=customer_id, sample_folder=tmp_folder, force=False
-    )
-    assert (tmp_folder.parent / sample.internal_id).exists()
-    assert not tmp_folder.exists()
-
-
-def test_get_available_samples_no_samples_avail(
-    analysis_store_trio,
-    customer_id: str,
-    external_data_api: ExternalDataAPI,
-    ticket: str,
-    tmpdir_factory,
-):
-    # GIVEN that the empty directory created does not contain any correct folders
-    tmp_dir_path: Path = Path(tmpdir_factory.mktemp("not_sample_id", numbered=False))
-    available_samples = external_data_api.get_available_samples(folder=tmp_dir_path, ticket=ticket)
-    # THEN the function should return an empty list
-    assert available_samples == []
-
-
-def test_checksum(fastq_file: Path):
-    """Tests if the function correctly calculates md5sum and returns the correct result."""
-    # GIVEN a fastq file with corresponding correct md5 file and a fastq file with a corresponding incorrect md5 file
-    bad_md5sum_file_path: Path = fastq_file.parent.joinpath("fastq_run_R1_001.fastq.gz")
-
-    # THEN a file with a correct md5 sum should return true
-    assert check_md5sum(file_path=fastq_file, md5sum="a95cbb265540a2261fce941059784fd1")
-
-    # THEN a file with an incorrect md5 sum should return false
-    assert not check_md5sum(
-        file_path=bad_md5sum_file_path, md5sum="c690b0124173772ec4cbbc43709d84ee"
-    )
-
-
-def test_extract_checksum(fastq_file: Path):
-    """Tests if the function successfully extract the correct md5sum."""
-
-    # Given a file containing a md5sum
-    md5sum_file = Path(f"{fastq_file.as_posix()}.md5")
-
-    # Then the function should extract it
-    assert extract_md5sum(md5sum_file=md5sum_file) == "a95cbb265540a2261fce941059784fd1"
+    # GIVEN a sample linked to a case set to hold
+    for case_links in sample_obj.links:
+        case_links.family.action = "hold"
+    # WHEN the function is called
```

Comment:
Let's extend `CaseActions` with "hold" and use it here
```suggestion
        case_links.family.action = CaseActions.HOLD
        
    # WHEN the function is called
```